src/lib.rs

```diff
@@ -795,6 +795,26 @@ pub trait Capture {
         R: Into<Self::Time>;
 }
 
+/// A single input capture channel / pin
+///
+/// See `Capture` for details
+#[cfg(feature = "unproven")]
+pub trait CapturePin {
+    /// Type  of value returned by capture
+    type Capture;
+    /// Enumeration of `Capture` errors
+    ///
+    /// Possible errors:
+    ///
+    /// - *overcapture*, the previous capture value was overwritten because it
+    ///   was not read in a timely manner
+    type Error;
+
+    /// "Waits" for a transition in the capture `channel` and returns the value
+    /// of counter at that instant
+    fn capture(&mut self) -> nb::Result<Self::Capture, Self::Error>;
```

Comment:
Thanks for the PR, I would prefer not to have two methods called `capture`, could this one be `capture_pin` or something to delineate between the two?

Reply:
I don't believe having both `Capture.capture()` and `CapturePin.capture()` causes any issues. Presently, `Pwm` and `PwmPin` methods all share the same names, such as `enable`, `disable`, and `get_duty`. The reason why this is OK is because `Pwm` and `PwmPin` will never be implemented on the same type, since one represents a collection of PWM channels while the other represents a single PWM channel. As such, trait method name conflicts will never occur. `Capture` and `CapturePin` follow the same reasoning, ensuring that the two `capture()` methods will never conflict.